src/connector/src/source/nats/mod.rs

```diff
@@ -49,6 +82,173 @@ pub struct NatsProperties {
     pub unknown_fields: HashMap<String, String>,
 }
 
+impl NatsProperties {
+    pub fn set_config(&self, c: &mut Config) {
+        self.nats_properties_consumer.set_config(c);
+    }
+}
+
+/// Properties for the async-nats library.
+/// See <https://docs.rs/async-nats/latest/async_nats/jetstream/consumer/struct.Config.html>
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, WithOptions)]
+pub struct NatsPropertiesConsumer {
+    #[serde(rename = "consumer.deliver_subject")]
+    pub deliver_subject: Option<String>,
+
+    #[serde(rename = "consumer.durable_name")]
+    pub durable_name: Option<String>,
+
+    #[serde(rename = "consumer.name")]
+    pub name: Option<String>,
+
+    #[serde(rename = "consumer.description")]
+    pub description: Option<String>,
+
+    #[serde(rename = "consumer.deliver_policy")]
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    pub deliver_policy: Option<String>,
+
+    #[serde(rename = "consumer.ack_policy")]
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    pub ack_policy: Option<String>,
```

Comment:
maybe we can try to deser it into Enum directly, serde can help us reject unrecognized str.

Reply:
I tried to do this, but unfortunately couldn't find a clean way to do it -- the enums from the NATS crate do not implement `FromStr`, so it doesn’t seem possible to deserialize directly.

Reply:
nevermind, I will try refactor it later. Thanks for your contribution.